### ISSUE TITLE
Replace "Create an account or sign in" modal with USWDS modal component

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#6256]
#### Describe the solution you'd like.

For several reasons, most importantly, for accessibility, we'd like to replace the "Create an account or sign in" modal with the [USWDS component](https://designsystem.digital.gov/components/modal/).

For a screenshot of this modal, see this [document](https://docs.google.com/document/d/1b1SMIoDCw_pAq7w0QN9B3hQyGB0wV-_QXkGASqyfqj8/edit?usp=sharing)

#### Additional Context

- This should be the large modal format.
- This modal is shown to guest users applying for program.
- This modal will be an overlay, with the current page in the background.
- This modal should continue to be dismissible.

We can utilize this [helper function](https://github.com/civiform/civiform/blob/88813618cdc44338c35986a11b34c387c6f60230/server/app/views/ViewUtils.java#L394), modifying it as needed or putting it into its own utility class.

**Comment by melanie-exygy** [comment]
This was completed as part of North Star work here:  [issue 7876](https://github.com/civiform/civiform/pull/7876).
